src/internal_events/file.rs

```diff
@@ -82,61 +82,71 @@ mod source {
     use std::{io::Error, path::Path, time::Duration};
 
     use file_source::FileSourceInternalEvents;
-    use metrics::counter;
+    use metrics::{counter, register_counter, Counter};
 
     use super::{FileOpen, InternalEvent};
     use crate::emit;
-    use vector_common::internal_event::{error_stage, error_type};
+    use vector_common::internal_event::{
+        error_stage, error_type, ByteSize, CountByteSize, InternalEventHandle,
+        RegisterInternalEvent,
+    };
+    use vector_common::registered_event;
+
+    registered_event!(
+        FileBytesReceived {
+            file: String,
+        } => {
+            received_bytes: Counter = register_counter!("component_received_bytes_total", "protocol" => "file", "file" => self.file.clone()),
+            file: String = self.file,
+        }
+
+        fn emit(&self, data: ByteSize) {
+            self.received_bytes.increment(data.0 as u64);
+            trace!(message = "Bytes received.", byte_size = %data.0, protocol = "file", file = %self.file);
+        }
+    );
 
     #[derive(Debug)]
-    pub struct FileBytesReceived<'a> {
-        pub byte_size: usize,
-        pub file: &'a str,
+    pub struct FileEventsReceived {
+        pub file: String,
     }
 
-    impl<'a> InternalEvent for FileBytesReceived<'a> {
-        fn emit(self) {
-            trace!(
-                message = "Bytes received.",
-                byte_size = %self.byte_size,
-                protocol = "file",
-                file = %self.file,
-            );
-            counter!(
-                "component_received_bytes_total", self.byte_size as u64,
-                "protocol" => "file",
-                "file" => self.file.to_owned()
-            );
+    impl RegisterInternalEvent for FileEventsReceived {
```

Comment:
Would still prefer you use `registered_event!` for this too.

Reply:
Fixed.